core/runtime/src/main/java/org/apache/isis/core/runtime/system/persistence/PersistenceSession.java

```diff
@@ -108,6 +106,14 @@ default ObjectAdapter adapterFor(final RootOid rootOid) {
     IsisConfiguration getConfiguration();
 
     PersistenceManager getPersistenceManager();
+    
+    /**
+     * @param pojo
+     * @param type
+     * @return String representing an object's id.
+     * @since 2.0.0-M2
+     */
+    String identifierFor(Object pojo, State type);
```

Comment:
as stated above the Oid.State enum or Type enum are exchangable, ultimately only one should remain, the other thrown away;